### ISSUE TITLE
Odoo 9 Error Creating an invoice

### ISSUE THREAD

**Issue by alacret** [odoo/odoo#10667]
After Confirming a Sales Order, i try to create an invoice and odoo give me this error:

Configuration error!
Could not find any account to create the invoice, are you sure you have a chart of account installed?

The steps that i follow were the following:
1. Install Sales
2. Install Accounting
3. Configuring Chart of accounts accoring to Generic Template
4. Create, and confirm a Sales ORder
5. Try to create an invoice from the Sales Order

Any one can help with this?

**Comment by aario** [comment]
Same on version 10.0. Looks like no body tries to create invoices in multi company mode so no body fixes the issue. I read the source code and found some clue why this happens:
        p = self.partner_id if not company_id else self.partner_id.with_context(force_company=company_id)
        type = self.type
        if p:
            rec_account = p.property_account_receivable_id
            pay_account = p.property_account_payable_id
            if not rec_account and not pay_account:
                action = self.env.ref('account.action_account_config')
                msg = _('Cannot find a chart of accounts for this company, You should configure it. \nPlease go to Account Configuration.')
                raise RedirectWarning(msg, action.id, _('Go to the configuration panel'))
So I first assigns p to partner or company ID. An ID is usually an Integer number starting from 1 to some million. Later it assumes that the partner_id or whatever which is loaded into P is an object so tries to get:
p.property_account_receivable_id
And:
p.property_account_payable_id
When it fails it complains that there is no chart of account configured for it.
I think it should first load the partner with partner_id provided. Then try to get the accounts out of it. Since this error is open from 2016 I feel like the project is dead and no body cares about bugs. While appreciating all the efforts Odoo community did/doing developing this huge software, will post a patch if in case I had time to fix it myself.

**Comment by aario** [comment]
I posted here how I fixed it here:
https://www.odoo.com/forum/help-1/question/sale-invoice-cannot-find-chart-of-accounts-for-the-company-121384#answer_122279
The only bugs/lack of features I could find were:

- The error message should tell customer that there is no chart of account for company AND there is no account receivable and payable for customer and also Income account for the product in the invoice. So the error message is not complete.

- There is no way to save selected chart of account for a company. Therefore I needed to run update query on database to do so.
